File: 128.longest-consecutive-sequence.py
#
# @lc app=leetcode id=128 lang=python
#
# [128] Longest Consecutive Sequence
#

# @lc code=start
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        map = set(nums)
        max_length = 0
            
        for num in map:
            if (num - 1) not in map:
                length = 1
                while (num + length) in map:
                    length += 1
                
                max_length = max(max_length, length)
        
        return max_length
        
# @lc code=end

File: 128.longest-consecutive-sequence.java
/*
 * @lc app=leetcode id=128 lang=java
 *
 * [128] Longest Consecutive Sequence
 */

// @lc code=start

import java.util.HashSet;
import java.util.Set;

class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for (int n : nums) set.add(n);
        
        int longest = 0;

        for(int n : set) {
            if (!set.contains(n - 1)) {
                int len = 1;
                while(set.contains(n + len)) len++;
                
                longest = Math.max(longest, len);
            }
        }

        return longest;
    }
}
// @lc code=end

